ignite/cmd/plugin.go

```diff
@@ -57,6 +61,97 @@ func UnloadPlugins() {
 	}
 }
 
+func linkPluginHooks(rootCmd *cobra.Command, p *plugin.Plugin) {
+	if p.Error != nil {
+		return
+	}
+
+	for _, hook := range p.Interface.Hooks() {
+		linkPluginHook(rootCmd, p, hook)
+	}
+}
+
+func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook plugin.Hook) {
+	cmdPath := hook.PlaceHookOn
+
+	if !strings.HasPrefix(cmdPath, "ignite") {
+		// cmdPath must start with `ignite ` before comparison with
+		// cmd.CommandPath()
+		cmdPath = "ignite " + cmdPath
+	}
+
+	cmdPath = strings.TrimSpace(cmdPath)
+
+	cmd := findCommandByPath(rootCmd, cmdPath)
+
+	if cmd == nil {
+		p.Error = errors.Errorf("unable to find commandPath %q for plugin hook %q", cmdPath, hook.Name)
+		return
+	}
+
+	if !cmd.Runnable() {
+		p.Error = errors.Errorf("can't attach plugin hook %q to non executable command %s", hook.Name, hook.PlaceHookOn)
+		return
+	}
+
+	preRun := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if preRun != nil {
+			err := preRun(cmd, args)
+			if err != nil {
+				return err
+			}
+		}
+
+		err := p.Interface.ExecuteHookPre(hook.Name, args)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	runCmd := cmd.RunE
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if runCmd != nil {
+			err := runCmd(cmd, args)
+
+			// if the command has failed the `PostRun` will not execute. here we execute the cleanup step before returnning.
+			if err != nil {
+				p.Interface.ExecuteHookCleanUp(hook.Name, args)
+			}
+
+			return err
+		}
+
+		time.Sleep(100 * time.Millisecond)
+		return nil
```

Comment:
This will execute the cleanup before the post, this is probably not what you want. Let's go back to the first implem where cleanup is executed in `PostRun`. I'm not sure we can address the problem that I mentioned [here](https://github.com/ignite/cli/pull/3011#discussion_r1009283197).

Reply:
I think this will only run if the original `RunE` implementation returns an error in this case the `PostRunE` will not invoke so executing `ExecuteHookCleanUp` will allow clean up from the `PreRun` hook. Unless my understanding of the command lifecycle in cobra is wrong.

Reply:
Since I rewrote the unit test, I can say you're right sorry for the confusion :)
Well done!